prow/metrics/http_test.go

```diff
@@ -56,3 +65,192 @@ func TestPowersOfTwoBetween(t *testing.T) {
+            testnamespace_http_request_duration_seconds_bucket{method="GET",path="",status="200",user_agent="",le="1"} %d
+            testnamespace_http_request_duration_seconds_bucket{method="GET",path="",status="200",user_agent="",le="2"} %d
+            testnamespace_http_request_duration_seconds_bucket{method="GET",path="",status="200",user_agent="",le="+Inf"} %d
+            testnamespace_http_request_duration_seconds_sum{method="GET",path="",status="200",user_agent=""} %f
+            testnamespace_http_request_duration_seconds_count{method="GET",path="",status="200",user_agent=""} %d
+			`,
+			expectedResponseSizeOut: `
+			# HELP testnamespace_http_response_size_bytes http response size in bytes
+            # TYPE testnamespace_http_response_size_bytes histogram
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="256"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="512"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="1024"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="2048"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="4096"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="8192"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="16384"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="32768"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="65536"} %d
+            testnamespace_http_response_size_bytes_bucket{method="GET",path="",status="200",user_agent="",le="+Inf"} %d
+            testnamespace_http_response_size_bytes_sum{method="GET",path="",status="200",user_agent=""} %d
+            testnamespace_http_response_size_bytes_count{method="GET",path="",status="200",user_agent=""} %d
+			`,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			httpResponseSize := HttpResponseSize(tc.namespace, 256, 65536)
+			httpRequestDuration := HttpRequestDuration(tc.namespace, 0.0001, 2)
+
+			simplifier := simplifypath.NewSimplifier(simplifypath.L(""))
+			handler := traceHandlerWithCustomTimer(simplifier, httpRequestDuration, httpResponseSize, tc.customTimer)(tc.dummyWriter)
+			rr := httptest.NewRecorder()
+			req, err := http.NewRequest("GET", "http://example.com", nil)
+			if err != nil {
+				t.Errorf("error while creating dummy request: %w", err)
+			}
+			handler.ServeHTTP(rr, req)
+			handleMetricsCalls++
```

Comment:
We construct everything per testcase so the value of this must always be `1` or not? It just happens to work now as we have exactly one testcase that increments the uninitialized var so it end sup being `1`.